llvm/include/llvm/Analysis/ValueTracking.h

```diff
@@ -124,11 +124,7 @@ bool isOnlyUsedInZeroEqualityComparison(const Instruction *CxtI);
 /// specified, perform context-sensitive analysis and return true if the
 /// pointer couldn't possibly be null at the specified instruction.
 /// Supports values with integer or pointer type and vectors of integers.
-bool isKnownNonZero(const Value *V, const DataLayout &DL, unsigned Depth = 0,
-                    AssumptionCache *AC = nullptr,
-                    const Instruction *CxtI = nullptr,
-                    const DominatorTree *DT = nullptr,
-                    bool UseInstrInfo = true);
```

Comment:
Do you also need to erase the old API? `computeKnownBits` has both.

Reply:
I don't see any reason to keep the old API. It is unlikely to break LLVM downstream users' build (except some static analysis tools).

Reply:
There are a few places it makes the code simpler (i.e the cases with just DL and depth==0).